src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java

```diff
@@ -0,0 +1,105 @@
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.IntervalArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.OpticalDuplicatesArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.OptionalIntervalArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSink;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.markduplicates.DuplicationMetrics;
+import org.broadinstitute.hellbender.utils.read.markduplicates.OpticalDuplicateFinder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+@CommandLineProgramProperties(
+        summary ="Marks duplicates on spark",
+        oneLineSummary ="Mark Duplicates",
+        programGroup = SparkProgramGroup.class)
+public final class MarkDuplicatesSpark extends SparkCommandLineProgram {
+    private static final long serialVersionUID = 1L;
+
+    //Bases below this quality will not be included in picking the best read from a set of duplicates.
+    private static final int MIN_BASE_QUAL = 15;
+
+    // Used to set an attribute on the GATKRead marking this read as an optical duplicate.
+    private static final String OPTICAL_DUPLICATE_TOTAL_ATTRIBUTE_NAME = "OD";
+
+    @Argument(doc = "uri for the input bam, either a local file path or a gs:// bucket path",
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            optional = false)
+    protected String bam;
+
+    @Argument(doc = "the output bam", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    protected String output;
+
+    @Argument(doc = "File to write duplication metrics to.", optional=true,
+            shortName = "M", fullName = "METRICS_FILE")
+    protected File metricsFile;
+
+    @ArgumentCollection
+    protected OpticalDuplicatesArgumentCollection opticalDuplicatesArgumentCollection = new OpticalDuplicatesArgumentCollection();
+
+    @ArgumentCollection
+    protected IntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
+    public static JavaRDD<GATKRead> mark(final JavaRDD<GATKRead> reads, final SAMFileHeader header,
```

Comment:
I think that's a good point. However, I'm in favor of keeping this as an RDD of reads (and not an RDD of pairs). The read oriented part of the pipeline is done after BQSR. After that, it's the haplotypecaller, and that doesn't care about pairs (IIRC).

@droazen , what are your thoughts?